src/main.cpp

```diff
@@ -794,7 +796,7 @@ bool AcceptToMemoryPool(CTxMemPool& pool, CValidationState &state, const CTransa
 
     // Rather not work on nonstandard transactions (unless -testnet/-regtest)
     string reason;
-    if (fRequireStandard && !IsStandardTx(tx, reason))
+    if (!policy.ApproveTx(tx, reason))
```

Comment:
It won't break in subtle ways: they will break hard and require rebase or they won't.
More of AcceptToMemoryPool can be moved to policy later and changing the "API" will be unavoidable if we want to start with something simple like this.
Another possibility is keep on complaianing about minor details of a trivial PR and wait another full year to have a policy class.
Please, if you don't like this PR exactly as it is you can code your particular nits instead of giving vague complains like this one.